test/e2e/azure_failuredomains.go

```diff
@@ -83,15 +82,9 @@ func AzureFailureDomainsSpec(ctx context.Context, inputGetter func() AzureFailur
 		// TODO: Find alternative when the number of zones is > 1 but doesn't equal to number of control plane machines.
 		if len(input.Cluster.Status.FailureDomains) == 3 {
 			By("Ensuring control planes are spread across availability zones.")
-			key := client.ObjectKeyFromObject(input.Cluster)
 			failureDomainsInput := framework.AssertControlPlaneFailureDomainsInput{
-				GetLister:  input.BootstrapClusterProxy.GetClient(),
-				ClusterKey: key,
-				ExpectedFailureDomains: map[string]int{
```

Comment:
Just to double check, do we no longer need to have `ExpectedFailureDomains` here?

Reply:
Correct, this is a documented change w/ 1.2.0. See this section:

- https://github.com/fabriziopandini/cluster-api/blob/main/docs/book/src/developer/providers/v1.1-to-v1.2.md#other

The "The AssertControlPlaneFailureDomains function in the E2E test framework has been modified..." part